### PR TITLE
Add kubectl-minio plugin

### PR DESCRIPTION
**Pull request by nitisht**
This PR adds support for commands

- [x] kubectl minio init
- [x] kubectl minio tenant create
- [x] kubectl minio tenant update
- [x] kubectl minio tenant scale
- [x] kubectl minio tenant delete 

Will send below pending items will in a separate PR

- [ ] kubectl minio pv create
- [ ] Support for `-o` flag to general corresponding yaml

### CODE REVIEW

File: kubectl-minio/cmd/cmd-tenant.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2020, MinIO, Inc.
```
Comment (harshavardhana) [inline kubectl-minio/cmd/cmd-tenant.go:2]: Copyright should mention operator @nitisht

File: kubectl-minio/cmd/constants/constants.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ */
+
+package constants
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// StaticYamlPath is path to the static yaml files
```
Comment (harshavardhana) [inline kubectl-minio/cmd/constants/constants.go:24]: club all the const together like 

```
const (
...
...
...
)
```

File: kubectl-minio/cmd/constants/constants.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const StaticYamlPath = "https://raw.githubusercontent.com/nitisht/kubectl-minio/master/cmd/static/static.yaml"
```
Comment (harshavardhana) [inline kubectl-minio/cmd/constants/constants.go:25]: not nice :-)

File: kubectl-minio/cmd/kubectl-minio.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+// NewCmdMinIO creates a new root command for kubectl-minio
+func NewCmdMinIO(streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewMinIOOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:          "minio",
+		Short:        "manipulate, inspect and troubleshoot MinIO Operator Specific CRDs.",
+		Long:         `MinIO Operator plugin can be used to manipulate, inspect and troubleshoot MinIO Operator Specific CRDs.`,
```
Comment (harshavardhana) [inline kubectl-minio/cmd/kubectl-minio.go:54]:
```suggestion
		Long:         `MinIO Operator plugin to manage MinIO operator CRDs`,
```

Simple

File: kubectl-minio/cmd/kubectl-minio.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	}
+}
+
+// NewCmdMinIO creates a new root command for kubectl-minio
+func NewCmdMinIO(streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewMinIOOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:          "minio",
+		Short:        "manipulate, inspect and troubleshoot MinIO Operator Specific CRDs.",
```
Comment (harshavardhana) [inline kubectl-minio/cmd/kubectl-minio.go:53]:
```suggestion
		Short:        "manage MinIO operator CRDs",
```

File: kubectl-minio/main.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+package main
+
+import (
+	"os"
+
+	"github.com/minio/kubectl-minio/cmd"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+var version = "DEVELOPMENT.GOGET"
```
Comment (harshavardhana) [inline kubectl-minio/main.go:27]: We don't need this just set it to `0.0.0` , we shall use semver here and even match this to operator release tags for now.

File: kubectl-minio/README.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+## Operator Deployment
+
+Command:
+
+`kubectl minio init [options] [flags]`
+
+Options:
+
+- `--image=minio/k8s-operator:2.0.6`
+- `--namespace-to-watch=default`
```
Comment (kerneltime) [inline kubectl-minio/README.md:25]: Operator should work across all namespaces?

File: kubectl-minio/cmd/constants/constants.go (hunk 3 above)
Comment (nitisht) [reply kubectl-minio/cmd/constants/constants.go:25]: Yeah agree, I want to add this yaml to the binary, looking at https://github.com/markbates/pkger

Comment (nitisht) [reply kubectl-minio/cmd/constants/constants.go:25]: Yeah agree, I want to add this yaml to the binary, looking at https://github.com/markbates/pkger

File: kubectl-minio/README.md (hunk 7 above)
Comment (nitisht) [reply kubectl-minio/README.md:25]: > Maybe a comma-separated list?

Currently the operator can either watch all namespaces or one namespace set in this `WATCH_NAMESPACE` env.

File: kubectl-minio/cmd/constants/constants.go (hunk 3 above)
Comment (nitisht) [reply kubectl-minio/cmd/constants/constants.go]: fixed

**Comment by nitisht** [comment]
@harshavardhana @kerneltime this PR is ready for review. PTAL

**Comment by nitisht** [comment]
Closing this as kustomize seems to be better approach.

**Pull request closed without merging**